service-base/src/test/java/org/eclipse/hono/service/credentials/BaseCredentialsServiceTest.java

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+package org.eclipse.hono.service.credentials;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.eclipse.hono.util.CredentialsConstants.FIELD_SECRETS;
+
+import org.eclipse.hono.config.ServiceConfigProperties;
+import org.eclipse.hono.util.CredentialsResult;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Unit tests for BaseCredentialsService
+ *
+ */
+public class BaseCredentialsServiceTest {
+
+    private BaseCredentialsService<ServiceConfigProperties> service;
+
+    private static final String TEST_FIELD = "dummy_test";
+
+    @Before
+    public void setUp() {
+        this.service = buildBaseCredentialsServiceDummy();
+    }
+
+    @Test
+    public void testIsValidTimestampForFieldWithValidTimestamps() {
+        String iso8601TimeStamp = "2007-04-05T12:30-02:00";
+
+        JsonObject testData = new JsonObject();
+        testData.put(TEST_FIELD, iso8601TimeStamp);
+        boolean result = service.isValidTimestampForField(testData, TEST_FIELD);
+        assertTrue(result);
+
+        String shortIso8601TimeStamp = "2007-04-05T14:30";
+        testData.put(TEST_FIELD, shortIso8601TimeStamp);
+        result = service.isValidTimestampForField(testData, TEST_FIELD);
+        assertTrue(result);
+    }
+
+    @Test
+    public void testIsValidTimestampForFieldWithInValidTimestamp() {
+        String iso8601TimeStamp = "yakshaver";
+
+        JsonObject testData = new JsonObject();
+        testData.put(TEST_FIELD, iso8601TimeStamp);
+
+        boolean result = service.isValidTimestampForField(testData, TEST_FIELD);
```

Comment:
"result" does not match well with many occurences of "valid" below. Can you find a better name?
E.g. always call them "result", or even consider to skip the temporary boolean variable.

Reply:
Aren't we just testing the correctness of the Java implementation of ISO8601 timestamps here?

Reply:
You're right, I will remove the unnecessary variable.

The test checks that our implementation of `isValidTimestampForField `returns true for a valid ISO8601 timestamp which is defined as time stamp format in the Credentials API. Therefore I would leave the test in place. If somebody changes the method, we might break our api.